### ISSUE TITLE
Inline <!-- markdownlint-enable --> overrides disabled rules in config/CLI

### ISSUE THREAD

**Issue by ekropotin** [igorshubovych/markdownlint-cli#561]
## Description

If a rule is disabled via the CLI (--disable) or configuration file (.markdownlint.yaml), it is still reported when the markdown file contains a  directive. This makes it impossible to fully disable a rule project-wide, because inline comments can re-enable it.

## Steps to reproduce

test.md:

```markdown
<!-- markdownlint-enable MD013 -->
This string is longer than 80 chars: In the labyrinth of distributed systems, every message is both a promise and a potential betrayal.
```

run:

```bash
markdownlint --disable MD013 -- test.md
```

## Expected behaviour

No violations reported, since MD013 was disabled explicitly via --disable.

## Actual behaviour

MD013 is still reported:

```bash
test.md:2:81 MD013/line-length Line length [Expected: 80; Actual: 135]
```

## Motivation / Rationale

- **Consistency with configuration intent:** A project maintainer or CI pipeline may choose to disable a rule globally (via config or CLI). This decision should not be overridden by individual file authors through inline directives, otherwise the global enforcement is undermined.
- **Analogy with other linters:** Tools like ESLint treat configuration as the highest authority. Inline directives can relax active rules but cannot bring back rules that were explicitly turned off in config. This avoids conflicts and surprises.
- **Principle of least astonishment:** When a user disables a rule in .markdownlint.yaml or with --disable, they reasonably expect it to stay disabled regardless of file contents. Having inline comments silently resurrect rules is unexpected and confusing.
- **Practical impact:** Teams may disable rules like MD013 (line length) globally for good reasons (e.g., preferring soft wrapping in editors). With current behavior, contributors who paste `<!-- markdownlint-enable MD013 -->` into a file unintentionally reintroduce warnings that CI or pre-commit hooks were meant to suppress.

## Notes

- This occurs whether the rule is disabled via .markdownlint.yaml or --disable MD013.
- A natural expectation is: Config/CLI disables have the highest precedence. Inline directives should only apply to rules that are enabled in the config.

**Comment by DavidAnson** [comment]
And also:

> **Warning** `/* eslint-enable */` without any specific rules listed causes all disabled rules to be re-enabled.

**Comment by ekropotin** [comment]
I agree, that ESLint doc line is a bit misleading out of context. Inline configuration comments do have the highest priority, but **only within the set of rules that are active in the config/CLI**. If a rule is turned off in `.eslintrc` (or via CLI), an inline `/* eslint-enable ... */` cannot bring it back to life, because ESLint never runs that rule in the first place.

I think, what the doc really means is: if a rule is enabled in config (warn/error), then inline comments can suppress or re-enable it, and those inline instructions take precedence over config severity.

I did some testing, and it confirms ESLint behaves this way.

test.js:

```javascript
/* eslint-disable no-console */
console.log("hello")
/* eslint-enable no-console */
console.log("world")
```

## `no-console` rule is enabled in config

```shell
>node_modules/.bin/eslint test.js
/tmp/eslinttest/test.js
  4:1  error  Unexpected console statement  no-console
```

## `no-console` rule is disabled in config

```shell
node_modules/.bin/eslint test.js
/tmp/eslinttest/test.js
  1:1  warning  Unused eslint-disable directive (no problems were reported from 'no-console')
```

This is why the current markdownlint behavior feels inconsistent:  can “resurrect” a rule even if it was disabled in config, whereas ESLint treats config as the absolute source of truth and only lets inline comments control enabled rules.

**Comment by DavidAnson** [comment]
Below is a self-contained example of using inline comments to enable an ESLint rule that is disabled in a configuration file (based on the current/latest version of ESLint). Specifically, note that removing the inline comment allows the configuration file's `off` setting to take effect for the `no-unused-vars` rule and produces no errors.

```bash
user@HOST cli-561 % cat package.json 
{
  "dependencies": {
    "@eslint/js": "^9.34.0",
    "eslint": "^9.34.0"
  },
  "scripts": {
    "lint": "eslint"
  }
}
user@HOST cli-561 % cat eslint.config.mjs 
import js from "@eslint/js";

export default [
  js.configs.all,
  {
    "rules": {
      "no-unused-vars": "off"
    }
  }
];

/* eslint no-unused-vars: "error" */

const unused = 0;
user@HOST cli-561 % npm run lint

> lint
> eslint


/Users/david/Documents/cli-561/eslint.config.mjs
  14:7  error  'unused' is assigned a value but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)

user@HOST cli-561 %
```

**Comment by ekropotin** [comment]
That's interesting!

Apparently, the enable/disable directives can’t resurrect a rule, but the **severity override comment can**.

So, if you try the same test with /* eslint-enable no-unused-vars */ (which is semantically close to `markdownlint-enable`), the effect will be different.

**Comment by DavidAnson** [comment]
Fundamentally, I think people are more used to local conditions overriding global conditions than the opposite. An easy example is speed limits: a city road might have a default limit of 40, but where there’s a sharp turn it drops to 30 and where there’s a school zone it drops to 20 and if there’s flooding in one area, it drops to 0. When thinking locally, there is additional context that’s more relevant to the situation and I claim it’s common for local policy to override broader policy.

That said, here are my thoughts on the motivation/rationale section above:

- “Consistency with configuration intent”
  - You make a distinction between maintainers and authors, but they’re all working on the same project. If a project wants to ban configuration overrides, that’s a policy it can decide on. But not all projects want this.
- “Analogy with other linters”
  - As shown above, ESLint allows local overrides of global configuration, so the behavior here is consistent. You might say both projects made the wrong decision, but I’d guess they used roughly the same logic.
- “Principle of least astonishment”
  - I claim that having inline comments sometimes work and sometimes not work (as proposed) would be more astonishing than having them work the same all the time like they do now.
- “Practical impact”
  - As above, if a project wants to define a policy to ban rule enablement, it can, and might enforce that via reviews. To implement stronger checks, CI could scan lint errors for instances of banned rules and fail if encountered.

**Comment by ekropotin** [comment]
Thanks for sharing your view, @DavidAnson 

 I don’t necessarily agree with everything, but I totally respect the maintainers’ right to run the project their way. I’m good with this issue being closed.
